index.bs

```diff
@@ -3459,7 +3548,109 @@ The {{MLInstanceNormalizationOptions}} members are:
 </details>
 </div>
 
-### The leakyRelu() method ### {#api-mlgraphbuilder-leakyrelu}
+### layerNormalization ### {#api-mlgraphbuilder-layernorm}
+Normalize the input using [[Layer-Normalization]]. Unlike [[#api-mlgraphbuilder-batchnorm]] where the mean and variance values are computed across all the samples in the batch dimension while the model is trained, and in [[#api-mlgraphbuilder-instancenorm]] where the mean and variance values are computed on the fly for each input feature of each individual sample in the batch, the means and variance values of the layer normalization are computed on the fly across all the input features of each individual sample in the batch.
+
+<script type=idl>
+dictionary MLLayerNormalizationOptions {
+  MLOperand scale;
+  MLOperand bias;
+  sequence<unsigned long> axes;
```

Comment:
Are there any restrictions on the `axes`? Because it would also indicate the shape of `scale` and `bias`, should the `axes` contain consecutive dimensions? 

For example, given a 4-D tensor, could `axes` be [0, 2, 3], [1, 3], [3, 2, 1], [2, 0] and so on?

Reply:
Restrictions would be the same as with `reduce`: no duplicates, all axes < rank, order agnostic.

Reply:
> Given an input shape of [1,2,3,4] and axes of [1,3], scale would have a shape of [2.4] (which is broadcast compatible with [1,2.3,4] ).

IIUC, scale in shape [2, 4] is not broadcast compatible with shape [1, 2, 3, 4]. Do you mean shape [2, 1, 4]? And if the axes is [3, 1], would the scale shape be [4, 2]? It is not broadcast compatible with shape [1, 2, 3, 4] either. Let me know if I misunderstood anything.

Reply:
@huningxin : Sorry, I meant the `scale` is broadcast-compatible with the `input` after being normalized (unsqueezed) from `[2,4]` back to `[1,2,1,4]`.

Reply:
@fdwr 

> @huningxin : Sorry, I meant above that the scale would be broadcast-compatible with the input after being rank-normalized (unsqueezed) from [2,4] back to [1,2,1,4].

It makes sense, thanks for the explanation.

> And if the axes is [3, 1], would the scale shape be [4, 2]? 

Then, I guess to handle this case, the algorithm step should sort the `axes` in ascending order before validating the scale and bias.

Reply:
> we expect the scale shape to be [4, 2] and let the implementation to permute and unsqueeze accordingly?

Prototyping this way: https://chromium-review.googlesource.com/c/chromium/src/+/5068275, resolving this comment now.